yt/frontends/flash/data_structures.py

```diff
@@ -191,25 +191,31 @@ def __init__(
 
         if self.particle_filename is None:
             # try to guess the particle filename
-            try:
-                self._particle_handle = HDF5FileHandler(
-                    filename.replace("plt_cnt", "part")
-                )
-                self.particle_filename = filename.replace("plt_cnt", "part")
-                mylog.info(
-                    "Particle file found: %s", self.particle_filename.split("/")[-1]
-                )
-            except OSError:
+            if "hdf5_plt_cnt" in filename:
+                # We have a plotfile, look for the particle file
+                try:
+                    pfn = filename.replace("plt_cnt", "part")
+                    self._particle_handle = HDF5FileHandler(pfn)
+                    self.particle_filename = pfn
+                    mylog.info(
+                        "Particle file found: %s", self.particle_filename.split("/")[-1]
+                    )
+                except OSError:
+                    self._particle_handle = self._handle
+            elif "hdf5_chk" in filename:
+                # This is a checkpoint file, should have the particles in it
                 self._particle_handle = self._handle
         else:
             # particle_filename is specified by user
             self._particle_handle = HDF5FileHandler(self.particle_filename)
 
         # Check if the particle file has the same time
         if self._particle_handle != self._handle:
-            part_time = self._particle_handle.handle.get("real scalars")[0][1]
-            plot_time = self._handle.handle.get("real scalars")[0][1]
-            if not np.isclose(part_time, plot_time):
+            part_time = self._particle_handle.handle.get("real scalars", None)
+            plot_time = self._handle.handle.get("real scalars", None)
+            if part_time is None:
+                raise RuntimeError("FLASH 2.x particle files are not supported!")
```

Comment:
I like it better without the explicit default value on `default`, but that's a nit.
```suggestion
            plot_time = self._handle.handle.get("real scalars")
            if (part_time := self._particle_handle.handle.get("real scalars")) is None:
                raise RuntimeError("FLASH 2.x particle files are not supported!")
```